var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -55,13 +55,16 @@ class Hdf5(AutotoolsPackage):
     variant('shared', default=True,
             description='Builds a shared version of the library')
 
-    variant('cxx', default=True, description='Enable C++ support')
-    variant('fortran', default=True, description='Enable Fortran support')
+    variant('unsupported', default=False,
+            description='Allow unsupported combinations of configure options')
+    variant('hl', default=False, description='Enable the high-level library')
+    variant('cxx', default=False, description='Enable C++ support')
```

Comment:
Why change to false?

Reply:
given the reply of @alalazo , let's add an in-code comment above along the lines
```
# +cxx+mpi is not explicitly supported, one can build HDF5 that way but there's no promise on its behavior
```

Reply:
As far as I understand, we have to introduce unnecessary constraints (~cxx) in the depends_on statement to enable thread safety (+threadsafe) even if we don't really care if the cxx interface is enabled or not. It might cause a conflict if there are two packages in a dependency tree one requiring +cxx, and another one requiring thread safety.

Reply:
> we have to introduce unnecessary constraints

If I understand you right, `conflicts('+threadsafe', when='+cxx')` is perfectly valid one and shall remain there assuming that 

> +cxx+threadsafe is claimed to be not compatible -> it's known to be broken

is still the case. We only remove it if proven otherwise. 

> if there are two packages in a dependency tree one requiring +cxx, and another one requiring thread safety.

you can't do anything about that. They simply require in-compatible variants. This shall be resolved upstream (hdf5).

Reply:
> They simply require incompatible variants.

One of the packages uses low-level C-API in multiple threads, another one uses CXX interface in a single thread. As far as I understand, both can get what they need from the same installation of the library if it's configured with +cxx+threadsafe.

Reply:
> both can get what they need from the same installation of the library if it's configured with +cxx+threadsafe.

then, feel free to add this comment and comment out the conflict statement so that no one tries to re-add it later.